rdr_service/tools/tool_libs/genomic_utils.py

```diff
@@ -2148,8 +2155,136 @@ def make_new_set():
         return set_dao.insert(new_set)
 
 
-def get_process_for_run(args, gcp_env):
+class UnblockSamples(ToolBase):
+    """
+    Unblocks a set of samples.
+    """
 
+    def __init__(self, args, gcp_env: GCPEnvConfigObject):
+        super().__init__(args, gcp_env)
+
+    def run(self):
+        self.gcp_env.activate_sql_proxy()
+        server_config = self.get_server_config()
+
+        member_dao = GenomicSetMemberDao()
+        metrics_dao = GenomicGCValidationMetricsDao()
+        aw1_raw_dao = GenomicAW1RawDao()
+        aw2_raw_dao = GenomicAW2RawDao()
+
+        no_aw1_data = []
+        no_aw2_data = []
+        needs_data_file_reconciliation = []
+
+        if self.args.file_path:
+            if not os.path.exists(self.args.file_path):
+                _logger.error(f'File {self.args.file_path} was not found.')
+                return 1
+        else:
+            _logger.error('Need --file-path')
+            return 1
+        if self.args.results_only is True and self.args.research_only is True:
+            _logger.error('Can only set one of --research-only or --results-only')
+            return 1
+
+        unblock_results = True
+        unblock_research = True
+        if self.args.results_only is True:
+            unblock_research = False
+        elif self.args.research_only is True:
+            unblock_results = False
+
+        with open(self.args.file_path, encoding='utf-8-sig') as file:
+            csv_reader = csv.reader(file)
+            id_list = []
+            header = next(csv_reader)[0]
+            if header == 'sample_id' or header == 'biobank_id':
+                id_type = header
+            else:
+                _logger.error('File header must be sample_id or biobank_id')
+                return 1
+
+            for row in csv_reader:
+                id_list.append(row[0])
+
+            if id_type == 'sample_id':
+                set_members = member_dao.get_members_from_sample_ids(id_list)  # will not return results in IGNORE state
+            else:
+                set_members = member_dao.get_members_from_biobank_ids(id_list)
+            for set_member in set_members:
+                if set_member.ignoreFlag == 1:
+                    continue
+                if unblock_results:
```

Comment:
You can get rid of the conditions 2190-2195 by changing 2217-2220 to the following:
`set_member.blockResults = 0
set_member.blockResearch = 0
if self.args.results_only: 
     set_member.blockResearch = 1
 if self.args.research_only: 
     set_member.blockResults = 1`

Reply:
There are records that have block_research = 1 and block_results = 0. If those were unblocked with research_only applied this logic would change block_results to 1.

Reply:
I see, that makes sense. I guess we do need it in the else then:
```if self.args.results_only: 
  set_member.blockResults = 0
elif self.args.research_only: 
  set_member.blockResearch = 0
else:
  set_member.blockResearch = 0
  set_member.blockResults = 0```